python/lsst/ip/isr/ptcDataset.py

```diff
@@ -519,68 +537,45 @@ def toTable(self):
-                              nSignalPoints*covMatrixSide**2).tolist(),
-                          'A_MATRIX_NO_B': np.array(self.aMatrixNoB[ampName]).reshape(
-                              covMatrixSide**2).tolist(),
-                          'FINAL_VARS': np.pad(np.array(self.finalVars[ampName]), (0, nPadPoints[ampName]),
-                                               'constant', constant_values=np.nan).tolist(),
-                          'FINAL_MODEL_VARS': np.pad(np.array(self.finalModelVars[ampName]),
-                                                     (0, nPadPoints[ampName]),
-                                                     'constant', constant_values=np.nan).tolist(),
-                          'FINAL_MEANS': np.pad(np.array(self.finalMeans[ampName]),
-                                                (0, nPadPoints[ampName]),
-                                                'constant', constant_values=np.nan).tolist(),
-                          'BAD_AMPS': np.array(self.badAmps).tolist() if len(self.badAmps) else np.nan,
-                          'PHOTO_CHARGE': np.array(self.photoCharge[ampName]).tolist(),
-                          } for ampName in self.ampNames])
+
+        badAmps = np.array(self.badAmps) if len(self.badAmps) else np.array([], dtype="U3")
+
+        catalogList = []
+        for ampName in self.ampNames:
+            ampDict = {
+                'AMPLIFIER_NAME': ampName,
+                'PTC_FIT_TYPE': self.ptcFitType,
+                'COV_MATRIX_SIDE': self.covMatrixSide,
+                'INPUT_EXP_ID_PAIRS': self.inputExpIdPairs[ampName],
+                'EXP_ID_MASK': self.expIdMask[ampName],
+                'RAW_EXP_TIMES': self.rawExpTimes[ampName],
+                'RAW_MEANS': self.rawMeans[ampName],
+                'RAW_VARS': self.rawVars[ampName],
+                'GAIN': self.gain[ampName],
+                'GAIN_ERR': self.gainErr[ampName],
+                'NOISE': self.noise[ampName],
+                'NOISE_ERR': self.noiseErr[ampName],
+                'PTC_FIT_PARS': np.array(self.ptcFitPars[ampName]),
+                'PTC_FIT_PARS_ERROR': np.array(self.ptcFitParsError[ampName]),
+                'PTC_FIT_CHI_SQ': self.ptcFitChiSq[ampName],
+                'PTC_TURNOFF': self.ptcTurnoff[ampName],
+                'A_MATRIX': self.aMatrix[ampName].ravel(),
+                'B_MATRIX': self.bMatrix[ampName].ravel(),
+                'A_MATRIX_NO_B': self.aMatrixNoB[ampName].ravel(),
+                'BAD_AMPS': badAmps,
+                'PHOTO_CHARGE': self.photoCharges[ampName],
+                'COVARIANCES': self.covariances[ampName].ravel(),
+                'COVARIANCES_MODEL': self.covariancesModel[ampName].ravel(),
+                'COVARIANCES_SQRT_WEIGHTS': self.covariancesSqrtWeights[ampName].ravel(),
+                'COVARIANCES_MODEL_NO_B': self.covariancesModelNoB[ampName].ravel(),
+                'FINAL_VARS': self.finalVars[ampName],
```

Comment:
Neither, really.  The construction of the dataset is such that for a given detector, all amps have the same input pairs and therefore are all filled the same (though it may be with `nan`s).  I believe that previously the reason that this did not work is because (a) we were filtering out bad values on read, which messed up the lengths per amp; (b) we were filtering out bad values on read inconsistently because of the astropy bug, which messed up the lengths within an amp.

Reply:
Ah yes, I left all of this part the same.  It's only the re-padding in the dataset that I removed.